app/src/main/java/com/ubergeek42/WeechatAndroid/utils/History.kt

```diff
@@ -0,0 +1,54 @@
+package com.ubergeek42.WeechatAndroid.utils
+
+import android.os.Bundle
+import android.text.Editable
+import android.widget.EditText
+import androidx.core.os.bundleOf
+import com.ubergeek42.WeechatAndroid.service.P
+
+class History {
+    private var index = -1
+    private var userInput: Editable? = null
+    private var selectionStart: Int? = null
+    private var selectionEnd: Int? = null
+
+    fun navigateOffset(editText: EditText, offset: Int) {
+        if (index == -1) {
+            userInput = editText.text
+            selectionStart = editText.selectionStart
+            selectionEnd = editText.selectionEnd
+        }
+        val newIndex = index + offset
+        messageAt(newIndex)?.let {
+            editText.text = it
+            editText.post {
+                if (newIndex == -1) {
+                    // Restore user selection.
+                    editText.setSelection(selectionStart ?: 0, selectionEnd ?: editText.length())
```

Comment:
Can `selectionStart` etc be `null` at this point? Why not making it non-nullable?

Reply:
Made everything non-null.